lib/core/src/connection_pool.cpp

```diff
@@ -125,21 +126,37 @@ namespace irods
                                      const int _port,
                                      std::optional<experimental::fully_qualified_username> _proxy_username,
                                      experimental::fully_qualified_username _username,
-                                     const int _refresh_time,
-                                     std::function<void(RcComm&)> _auth_func)
+                                     std::function<void(RcComm&)> _auth_func,
+                                     const connection_pool_options& _options)
         : host_{_host}
         , port_{_port}
         , proxy_username_{std::move(_proxy_username)}
         , username_{std::move(_username)}
-        , refresh_time_(_refresh_time)
         , auth_func_{std::move(_auth_func)}
         , conn_ctxs_(_size)
+        , options_{_options}
     {
         if (_size < 1) {
             // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-array-to-pointer-decay)
             THROW(SYS_INVALID_INPUT_PARAM, "Invalid connection pool size");
         }
 
+        if (_options.number_of_retrievals_before_connection_refresh) {
+            if (*_options.number_of_retrievals_before_connection_refresh <= 0) {
+                // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-array-to-pointer-decay)
+                THROW(SYS_INVALID_INPUT_PARAM,
+                      "Value for option [refresh_connections_after_n_retrievals] must be greater than zero");
```

Comment:
update to 'number_of_retrievals_before_connection_refresh'

Reply:
done